src/Terminal/commands/grep.ts

```diff
@@ -166,86 +153,100 @@ class Args {
-      if (allArgs.includes(fullArg)) {
-        options[key as keyof Options] = true;
-        isOption = true;
-      }
-      // check multiflag args
-      const multiFlag = stripDash(fullArg);
-      const shortArgs = theseArgs.short.map(stripDash);
-      if (multiFlag.length > 1 && shortArgs.some((arg) => [...multiFlag].includes(arg))) {
-        options[key as keyof Options] = true;
-        isOption = true;
-      }
-    }
-    return [options, isOption];
-  }
-
-  splitOptsAndArgs(): [Options, string[], string, string, string] {
-    let outFile, limit, context;
-
-    [outFile, this.args] = this.spliceOptParam(VALID_ARGS.isToFile);
-    [limit, this.args] = this.spliceOptParam(VALID_ARGS.isMaxMatches);
-    [context, this.args] = this.spliceOptParam(VALID_ARGS.isPreContext);
-    if (!context) [context, this.args] = this.spliceOptParam(VALID_ARGS.isContext);
-    if (!context) [context, this.args] = this.spliceOptParam(VALID_ARGS.isPostContext);
-
-    const [options, otherArgs] = this.args.reduce(
-      ([options, otherArgs]: [Options, string[]], fullArg: string): [Options, string[]] => {
-        let isOption = false;
-        [options, isOption] = this.mapArgToOpts(fullArg, options);
-        return isOption ? [options, otherArgs] : [options, [...otherArgs, fullArg]];
-      },
-      [this.initOptions, []],
-    );
-    const outFileStr = outFile ?? "";
-    const limitNum = limit ?? "";
-    const contextNum = context ?? "";
-
-    return [options, otherArgs, outFileStr, contextNum, limitNum];
-  }
+  INIT_PARAMS: Params = {
+    isPreContext: "",
+    isContext: "",
+    isPostContext: "",
+    isMaxMatches: "",
+    isToFile: "",
+  };
 
-  spliceOptParam(validArgs: ArgStrings): [string, string[]] | [undefined, string[]] {
+  private spliceParam(validArgs: ArgStrings): string {
     const argIndex = [...validArgs.long, ...validArgs.short].reduce((ret: number, arg: string) => {
       const argIndex = this.args.indexOf(arg);
       return argIndex > -1 ? argIndex : ret;
     }, NaN);
 
-    if (isNaN(argIndex)) return [undefined, this.args];
+    if (isNaN(argIndex)) return "";
+
+    return this.args.splice(argIndex + 1, 1)[0];
+  }
+
+  private spliceOptionalParams(): Args {
+    for (const [key, args] of Object.keys(this.params).map((key): [string, ArgStrings] => [
+      key,
+      VALID_ARGS[key as keyof Options],
+    ])) {
+      this.params[key as keyof Params] = this.spliceParam(args);
+    }
+    return this;
+  }
+
+  private reduceToOptionsAndFiles(): string[] {
+    const stripDash = (arg: string) => arg.slice(1);
+    const argKeys = Object.keys(VALID_ARGS).map((k) => k as keyof Options);
+    const [allValidArgs, validFlagChars] = argKeys.reduce(
+      (ret: [readonly string[], string], key: keyof Options): [readonly string[], string] => {
+        const argStrings = VALID_ARGS[key];
+        return [
+          [...ret[0], ...argStrings.long, ...argStrings.short],
```

Comment:
Oh yeah, especially since you're doing the O(n^2) thing with list re-concatenation here again.

Reply:
I did recognise this repeated smelly-pattern from before, but figured it was acceptable in this instance as these arrays are fixed and short. 

It will be tweaked RE the above review though :)